src/src/com/microsoft/aad/adal/HttpWebRequest.java

```diff
@@ -93,7 +94,7 @@ private HttpURLConnection setupConnection() throws IOException {
             throw new IllegalArgumentException("requestURL");
         }
         HttpURLConnection.setFollowRedirects(true);
-        final HttpURLConnection connection = (HttpURLConnection)mUrl.openConnection();
+        final HttpURLConnection connection = HttpUrlConnectionFactory.createHttpUrlConnection(mUrl);
```

Comment:
you don't need this factory, you can use mockito to mock Url.openConnection method